lib/bandit/telemetry.ex

```diff
@@ -60,7 +60,27 @@ defmodule Bandit.Telemetry do
 
   The following events may be emitted within this span:
 
-  * `[:bandit, :request, :exception]`
+  * `[:bandit, :request, :error]`
+
+      The request for this span ended unexpectedly
+
+      This event contains the following measurements:
+
+      * `monotonic_time`: The time of this event, in `:native` units
+
+      This event contains the following metadata:
+
+      * `telemetry_span_context`: A unique identifier for this span
+      * `connection_telemetry_span_context`: The span context of the Thousand Island `:connection`
+        span which contains this request
+      * `conn`: The `Plug.Conn` representing this connection. Not present in cases where `error`
+        is also set and the nature of error is such that Bandit was unable to successfully build
+        the conn
+      * `kind`: The kind of unexpected condition, typically `:error`, `:throw` or `:exit`
+      * `reason`: The exception, throw value or exit reason which caused this unexpected termination
+      * `stacktrace`: The stacktrace of the location which caused this unexpected termination
+
+  * `[:bandit, :request, :exception]` (DEPRECATED)
```

Comment:
I realize that `exception` is (and always has been) the wrong term here, but I've deliberately been using it to mirror the naming conventions of [:telemetry.span/3](https://hexdocs.pm/telemetry/telemetry.html#span/3).

Reply:
Gotcha.

I only introduced a new event name just to not break the shape of the actual one when supporting more `kind`s.

Updating back to only `[:bandit, :request, :exception]`, with the correct `kind` and `reason` field, and maybe keeping the `exception` field in case `is_exception(reason)`, so that this doesn't break peoples telemetry handlers listening for that field.

It will break anyone matching on `kind: :exit` and `exception: exception`. This will now be corrected to `kind: :error` and `exception: exception`.

Reply:
That's fine